zuul-core/src/main/java/com/netflix/zuul/message/ZuulMessageImpl.java

```diff
@@ -178,6 +178,13 @@ public Iterable<HttpContent> getBodyContents() {
         return Collections.unmodifiableList(bodyChunks);
     }
 
+    @Override
+    public void resetBodyReader() {
+        for (final HttpContent chunk : bodyChunks) {
```

Comment:
Is it worth only resetting this when we know we have a body, i.e. effectively `setBody` has been called?

Reply:
Looking at the _soft contract_ we don't set the `hasBody` boolean in the `setBody` method. So going to not check that here as the hasBody doesn't determine any logic within the message impl currently.